hooks/pre-commit

```diff
@@ -0,0 +1,34 @@
+#!/bin/bash
+
+# pre-commit hook for git which builds the docker image for a given cloud-sdk version.
+# Copy into .git/hooks/  folder
```

Comment:
well, the only people its relevant for are the maintainers of the image so i don't think its that much of trouble to move it to that folder.  

re: update.sh   i like the precommit hook better as it doens't even allow you to commit if the build is flawed.   and anyone can execute outside of that by running
bash hooks/pre-commit